test/functional/action-params.spec.ts

```diff
@@ -77,49 +73,13 @@ describe("action parameters", () => {
     });
 
     before(() => {
-
         // reset metadata args storage
         defaultMetadataArgsStorage().reset();
 
 
-        @Middleware()
-        class SessionMiddleware implements MiddlewareInterface {
-            public use (requestOrContext: any, responseOrNext: any, next?: (err?: any) => any): any {
-                if (next) {
-                    return this.expSession(requestOrContext, responseOrNext, next);
-                } else {
-                    if (!this.koaSession) {
-                        this.koaSession = convert(KoaSession(requestOrContext.app));
-                    }
-                    return this.koaSession(requestOrContext, responseOrNext);
-                }
-            }
-
-            private expSession = session({
-                secret: "19majkel94_helps_pleerock",
-            });
-
-            private koaSession: any;
-        }
-
-        class User {
-            public username: string;
-            public location: string;
-            public twitter: string;
-        }
-
-        @Middleware()
-        class SetStateMiddleware implements MiddlewareInterface {
-            public use (context: any, next: (err?: any) => Promise<any>): Promise<any> {
-                const user = new User();
-                user.username = "pleerock";
-                user.location = "Dushanbe, Tajikistan";
-                user.twitter = "https://twitter.com/pleerock";
-                context.state = user;
-                return next().then(() => { console.log("here3", context.state); });
-            }
-        }
 
+        const {SetStateMiddleware} = require("../fakes/global-options/koa-middlewares/SetStateMiddleware");
```

Comment:
Does importing middlewares works even with `defaultMetadataArgsStorage().reset();`?
Why JS require instead of import?